mirrord-layer/src/socket/ops.rs

```diff
@@ -210,167 +275,36 @@ pub(super) fn connect(sockfd: RawFd, remote_address: SocketAddr) -> HookResult<(
-            let MirrorAddress(mirror_address) = mirror_rx.blocking_recv()??;
-
-            let connect_to = SockAddr::from(mirror_address);
-
-            // Connect to the interceptor socket that is listening.
-            let connect_result =
-                unsafe { FN_CONNECT(sockfd, connect_to.as_ptr(), connect_to.len()) };
-
-            let err_code = errno::errno().0;
-            if connect_result == -1 && err_code != libc::EINPROGRESS && err_code != libc::EINTR {
-                error!(
-                    "connect -> Failed call to libc::connect with {:#?} errno is {:#?}",
-                    connect_result,
-                    errno::errno()
-                );
-                return Err(io::Error::last_os_error())?;
+        SocketKind::Tcp(_) => match user_socket_info.state {
+            SocketState::Initialized if !((is_ignored_port(port)) && (is_ignored_ip(ip))) => {
+                connect_outgoing::<TCP>(sockfd, remote_address, user_socket_info)
             }
+            SocketState::Initialized if !enabled_tcp_outgoing => Err(HookError::Bypass),
+            SocketState::Bound(Bound { address, .. }) => {
+                trace!("connect -> SocketState::Bound {:#?}", user_socket_info);
 
-            // Warning: We're treating `EINPROGRESS` as `Connected`!
-            let connected = Connected {
-                remote_address,
-                mirror_address,
-            };
-
-            Arc::get_mut(&mut user_socket_info).unwrap().state = SocketState::Connected(connected);
-
-            Ok::<(), HookError>(())
-        }
-        SocketState::Bound(Bound { address, .. }) => {
-            trace!("connect -> SocketState::Bound {:#?}", user_socket_info);
-
-            let address = SockAddr::from(address);
-            let bind_result = unsafe { FN_BIND(sockfd, address.as_ptr(), address.len()) };
+                let address = SockAddr::from(address);
+                let bind_result = unsafe { FN_BIND(sockfd, address.as_ptr(), address.len()) };
 
-            if bind_result != 0 {
-                error!(
+                if bind_result != 0 {
+                    error!(
                     "connect -> Failed to bind socket result {:?}, address: {:?}, sockfd: {:?}!",
                     bind_result, address, sockfd
                 );
 
-                Err(io::Error::last_os_error())?
-            } else {
-                let rawish_remote_address = SockAddr::from(remote_address);
-                let result = unsafe {
-                    FN_CONNECT(
-                        sockfd,
-                        rawish_remote_address.as_ptr(),
-                        rawish_remote_address.len(),
-                    )
-                };
-
-                if result != 0 {
                     Err(io::Error::last_os_error())?
                 } else {
-                    Ok::<_, HookError>(())
+                    Err(HookError::Bypass)
```

Comment:
What does this error mean?

Reply:
This was a close relative of Windows' "Error(Success)".